admin/src/js/controllers/edit-user.js

```diff
@@ -270,35 +298,41 @@ angular.module('controllers').controller('EditUserCtrl',
       $scope.setProcessing();
       $scope.errors = {};
       computeFields();
-      if (validateName() &&
-          validateRole() &&
-          validateContactAndFacility() &&
-          validatePasswordForEditUser()) {
+      if (
+        validateName() &&
+        validateRole() &&
+        validateContactAndFacility() &&
+        validatePasswordForEditUser()
+      ) {
         validateContactIsInPlace()
           .then(function(valid) {
             if (!valid) {
               $scope.setError();
               return;
             }
             changedUpdates($scope.editUserModel).then(function(updates) {
-              $q.resolve().then(function() {
-                if (!haveUpdates(updates)) {
-                  return;
-                } else if ($scope.editUserModel.id) {
-                  return UpdateUser($scope.editUserModel.username, updates);
-                } else {
-                  return CreateUser(updates);
-                }
-              })
+              $q.resolve()
+                .then(function() {
+                  if (!haveUpdates(updates)) {
+                    return;
+                  } else if ($scope.editUserModel.id) {
+                    return UpdateUser($scope.editUserModel.username, updates);
+                  } else {
+                    return CreateUser(updates);
+                  }
+                })
                 .then(function() {
                   $scope.setFinished();
                   // TODO: change this from a broadcast to a changes watcher
                   //       https://github.com/medic/medic-webapp/issues/4094
-                  $rootScope.$broadcast('UsersUpdated', $scope.editUserModel.id);
+                  $rootScope.$broadcast(
+                    'UsersUpdated',
+                    $scope.editUserModel.id
+                  );
                   $uibModalInstance.close();
                 })
                 .catch(function(err) {
-                  $scope.setError(err, 'Error updating user');
+                  $scope.setError(err, err.data.error);
```

Comment:
The error message is translated in the modal template but I think the `err.data.error` from API is just a plain text string. This means if you view the admin app in say Swahili the error message will still show in English. To support translation you could add translation keys to the response from API and use that here.